### DISCUSSION TITLE
How do I register transient views

### DISCUSSION THREAD

**Discussion by chenzanyu** [AvaloniaCommunity/Prism.Avalonia#103] [Q&A] [answered]
## Description
Navigation views are singleton by default. Is there any way to instantiate the ViewModel each time you navigate to a specified view？

## Environment
* **OS:**  _Windows 10, Ubuntu 22.04, etc._
*  **Prism.Avalonia Version:**
* **Avalonia Version:** 

## Severity (1-5)
1=Low (_annoyance_), 5=High (_crashes visual studio_)

## Steps To Reproduce
2. Click on '...'
3. Scroll down to '...'
4. Error message '...'

**Comment by DamianSuess** [comment]
@chenzanyu In the future, please keep all questions/inquiries in the **Discussions** section, as this is not an issue.

**Comment by DamianSuess** [comment, marked as answer]
Please read over the Prism Library's documentation on [Registering Types](https://prismlibrary.com/docs/dependency-injection/registering-types.html).

https://prismlibrary.com/docs/dependency-injection/registering-types.html
